src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BreakpointAllele.java

```diff
@@ -51,20 +68,56 @@ private InversionType getInversionType(final boolean fiveToThree, final boolean
         return InversionType.INV_NONE;
     }
 
+
     @Override
     public boolean equals(final Object o) {
         if (this == o) return true;
         if (o == null || getClass() != o.getClass()) return false;
         final BreakpointAllele that = (BreakpointAllele) o;
-        return inversionType == that.inversionType &&
-                Objects.equals(leftAlignedLeftBreakpoint, that.leftAlignedLeftBreakpoint) &&
-                Objects.equals(leftAlignedRightBreakpoint, that.leftAlignedRightBreakpoint) &&
-                Objects.equals(insertedSequence, that.insertedSequence) &&
-                Objects.equals(homology, that.homology);
+
+        if (leftAlignedLeftBreakpoint != null ? !leftAlignedLeftBreakpoint.equals(that.leftAlignedLeftBreakpoint) : that.leftAlignedLeftBreakpoint != null)
+            return false;
+        if (leftAlignedRightBreakpoint != null ? !leftAlignedRightBreakpoint.equals(that.leftAlignedRightBreakpoint) : that.leftAlignedRightBreakpoint != null)
+            return false;
+        if (insertedSequence != null ? !insertedSequence.equals(that.insertedSequence) : that.insertedSequence != null)
+            return false;
+        if (homology != null ? !homology.equals(that.homology) : that.homology != null) return false;
+        return inversionType == that.inversionType;
+
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(leftAlignedLeftBreakpoint, leftAlignedRightBreakpoint, insertedSequence, homology, inversionType);
+        int result = leftAlignedLeftBreakpoint != null ? leftAlignedLeftBreakpoint.hashCode() : 0;
```

Comment:
I have a favor to ask, too:  To avoid a separate PR for a trivial one-line change, could you repair the same non-deterministic behavior in `BreakpointEvidence.hashCode`?
Change line 156 in BreakpointEvidence.java from
`        result = mult * result + templateEnd.hashCode();`
to
`        result = mult * result + templateEnd.ordinal();`